Implement `emptyText` and `emptyValue` in `AutocompleteInput` (#8162)
marmelab/react-admin: docs/AutocompleteInput.md

See [Using in a `ReferenceInput>`](#using-in-a-referenceinput) below for more information.

## `emptyValue`

> An empty choice is always added (with a default `''` value, which you can overwrite with the `emptyValue` prop) on top of the options. You can furthermore customize the empty choice by using the `emptyText` prop, which can receive string or a React Element, which doesn't receive any props.

Reviewer: ```suggestion
An empty choice is always added (with a default `''` value, which you can override with the `emptyValue` prop) on top of the options. You can furthermore customize the empty choice by using the `emptyText` prop, which can receive a string or a React Element.
```

```jsx
<AutocompleteInput
    source="author_id"

>     emptyValue=""

Reviewer: ```suggestion
    emptyValue={0}
```

    emptyText="No author"
    choices={[
        { id: 123, name: 'Leo Tolstoi' },
        { id: 456, name: 'Jane Austen' },
    ]}
/>
```

> **Note**: `emptyValue` can not be set to `undefined` or `null` since the `dataProvider` method will recibe an empty string on submit due to the nature of HTML inputs.

Reviewer: ```suggestion
**Note**: `emptyValue` cannot be set to `undefined` or `null` since the `dataProvider` method will receive an empty string on submit due to the nature of HTML inputs.
```
